dist/expected.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dist
+
+import "github.com/gonum/quad"
+
+// ExpectedFixed numerically computes the expected value of the input function
+// under the probability distribution specified by the Quantiler. That
+// is, ExpectedFixed computes
+//
+//  ev ≈ E[f] = int_-inf^inf f(x) p(x) dx
+//
+// using the number of quadrature locations specified by evals. Concurrent
+// sets the number of allowed concurrent evaluations of f.
+//
+// ExpectedFixed is a wrapper around quad.Fixed. See the documentation for more
+// information.
+func ExpectedFixed(f func(float64) float64, q Quantiler, evals, concurrent int) (ev float64) {
```

Comment:
The subintegrals can produce different rounding.  You might have values [A, B, C, D], and with the sync version you would perform A/4 + B/4 + C/4 + D/4, but in the concurrent version (with 2 values) you perform (A/4 + B/4) + (C/4 + D/4).  Either way would have some rounding, but not the same.  Take  [1e30, 1e30, -2e30, 2.0] - the first method will produce an answer of .5, the second will produce an answer of 0.0.

The thing that would worry me is that the output would depend on the scheduling of goroutines, so different evaluations of the same ExpectedFixed(...) with concurrency would not always be the same if there were more than 2 subintegrals.